File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/Delphi/ViewCardSimpleDemo/ViewCardSimpleDemoData.pas
unit ViewCardSimpleDemoData;

interface

uses
  SysUtils, Classes, DB, cxStyles, cxGridCardView, cxClasses,
  cxGridTableView, Forms, DBClient, MidasLib;

type
  TViewCardSimpleDemoMainDM = class(TDataModule)
    tlbDEPARTMENTS: TClientDataSet;
    dsDEPARTMENTS: TDataSource;
    dsUSERS: TDataSource;
    tlbUSERS: TClientDataSet;
    tlbDEPARTMENTSID: TAutoIncField;
    tlbDEPARTMENTSNAME: TStringField;
    procedure DataModuleCreate(Sender: TObject);
  end;

var
  ViewCardSimpleDemoMainDM: TViewCardSimpleDemoMainDM;

implementation

{$R *.dfm}

procedure TViewCardSimpleDemoMainDM.DataModuleCreate(Sender: TObject);
var
  APath: string;
begin
  APath := ExtractFilePath(Application.ExeName) + '..\..\Data\';
  tlbDEPARTMENTS.LoadFromFile(APath + 'Departments.xml');
  tlbUSERS.LoadFromFile(APath + 'Users.xml');
end;

end.

File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/CBuilder/ViewCardSimpleDemo/ViewCardSimpleDemoData.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "ViewCardSimpleDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxGridCardView"
#pragma link "cxGridTableView"
#pragma link "cxStyles"
#pragma resource "*.dfm"
TViewCardSimpleDemoMainDM *ViewCardSimpleDemoMainDM;
//---------------------------------------------------------------------------
__fastcall TViewCardSimpleDemoMainDM::TViewCardSimpleDemoMainDM(TComponent* Owner)
  : TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TViewCardSimpleDemoMainDM::DataModuleCreate(TObject *Sender)
{
	String APath = ExtractFilePath(Application->ExeName) + "..\\..\\Data\\";
	tlbDEPARTMENTS->LoadFromFile(APath + "Departments.xml");
	tlbUSERS->LoadFromFile(APath + "Users.xml");
}
//---------------------------------------------------------------------------
